test/helpers/envtest.go

```diff
@@ -165,7 +165,7 @@ func NewTestEnvironment() *TestEnvironment {
 		return (&webhooks.VSphereFailureDomainWebhook{}).SetupWebhookWithManager(mgr)
 	}
 
-	mgr, err := manager.New(managerOpts)
+	mgr, err := manager.New(context.Background(), managerOpts)
```

Comment:
```suggestion
	mgr, err := manager.New(ctrl.SetupSignalHandler(), managerOpts)
```
It's a bit hard to tell but in CAPI we are also using this one. Should lead to a better cleanup

Reply:
Sure, let's go with it.

Reply:
With this change, need to update NewTestEnvironment() to consume an existed ctx (initialized in controllers/controller_suite_test.go using ctrl.SetupSingalHandler()), as we cannot initialized ctrl.SetupSingalHandler() for multiple times.

Reply:
Sounds good, thx!